src/webgl/p5.RendererGL.js

```diff
@@ -83,6 +81,93 @@ p5.RendererGL.prototype._initContext = function() {
     throw new Error(er);
   }
 };
+
+//This is helper function to reset the context anytime the attributes
+//are changed with setAttributes()
+
+p5.RendererGL.prototype._resetContext = function(attr, options, callback) {
+  var w = this.width;
+  var h = this.height;
+  var defaultId = 'defaultCanvas0';
+  var c = document.getElementById(defaultId);
+  if(c){
+    c.parentNode.removeChild(c);
+  }
+  c = document.createElement('canvas');
+  c.id = defaultId;
+  if (this._pInst._userNode) {
+    this._pInst._userNode.appendChild(c);
+  } else {
+    document.body.appendChild(c);
+  }
+  this._pInst.canvas = c;
+  this._pInst._setProperty('_renderer', new p5.RendererGL(this._pInst.canvas,
+    this._pInst, true, attr));
+  this._pInst._isdefaultGraphics = true;
+  this._pInst._renderer.resize(w, h);
+  this._pInst._renderer._applyDefaults();
+  this._pInst._elements.push(this._renderer);
+  if(typeof callback === 'function') {
+    //setTimeout with 0 forces the task to the back of the queue, this ensures that
+    //we finish switching out the renderer
+    setTimeout(function() {
+      callback.apply(window._renderer, options);
+    }, 0);
+  }
+};
+
+
+/**
+ *
+ * Set attributes for the WebGL Drawing context.
+ * This should be put in setup()
+ * The available attributes are
+ * alpha - indicates if the canvas contains an alpha buffer
+ * default is true
+ *
+ * depth - indicates whether the drawing buffer has a depth buffer
+ * of at least 16 bits - default is true
+ *
+ * stencil - indicates whether the drawing buffer has a stencil buffer
+ * of at least 8 bits
+ *
+ * antialias - indicates whether or not to perform anti-aliasing
+ * default is false
+ *
+ * premultipliedAlpha - indicates that the page compositor will assume
+ * the drawing buffer contains colors with pre-multiplied alpha
+ * default is false
+ *
+ * preserveDrawingBuffer - if true the buffers will not be cleared and
+ * and will preserve their values until cleared or overwritten by author
+ * (note that p5 clears automatically on draw loop)
+ * default is true
+ *
+ * @method setAttributes
+ * @param  {String|Object}  String name of attribute or object with key-value pairs
+ * @param  {Boolean}        New value of named attribute
+ *
+ */
+
+p5.prototype.setAttributes = function() {
+  if(!this._renderer.isP3D) {
+    console.log('setAttributes() only works in WebGL');
+    return;
+  }
+  var attr = {};
+  if(arguments.length === 2) {
+    attr[arguments[0]] = arguments[1];
+  }
+  else if (arguments.length === 1) {
+    attr = arguments[0];
+  } else {
```

Comment:
this case should eventually be handled by the friendly errors, so we don't need to build it into the logic. @almchung can we decide on a comment tag that is easily searchable to mark places that should be revisited when friendly error system is working? maybe something like "@TODO FES"?